Library/SPIClient.m

```diff
@@ -2031,4 +2090,31 @@ - (void)didReceiveError:(NSError *)error {
     });
 }
 
+#pragma mark - Internals for Validations
+
+- (NSString *)validatePosId:(NSString *)posId {
+    if (posId.length  > 16) {
+        posId = [posId substringWithRange:NSMakeRange(0, 16)];
+        NSLog(@"The Pos Id should be equal or less than 16 characters! It has been truncated");
+    }
+    
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:regexItemsForPosId options:NSRegularExpressionCaseInsensitive error:nil];
```

Comment:
As with Java, `NSRegularExpression` is thread-safe and should be reused.

Reply:
Fixed